app/controllers/api/v1/keywords_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class KeywordsController < ApplicationController
+      include Keywords
+
+      before_action :doorkeeper_authorize!
+      before_action :load_user
```

Comment:
Since the steps to authorize a request & get the current user are common in all APIs, so even though in this project, just the Keyword APIs request that but to make it as a generic solution & reusable, how about moving these parts to a concern, sth like:
```ruby
# app/controllers/concerns/api/v1/authentication.rb
module Api
  module V1
    module Authentication
  		extend ActiveSupport::Concern

  		included do
  			attr_reader :current_user

  			before_action :doorkeeper_authorize!
  			before_action :load_current_user!
			end

			private

			def load_current_user
				user_id = doorkeeper_token.resource_owner_id
				@current_user = user_id.present? User.find(doorkeeper_token.resource_owner_id) : nil
			end
    end
  end
end

# and in the ApplicationController
module Api
  module V1
    class ApplicationController < ActionController::API
    	include Api::V1::Authentication
  	end
	end
end

# KeywordController
module Api
  module V1
    class KeywordsController < ApplicationController
    ...
      def index
        render json: KeywordSerializer.new(search_keyword(current_user)).serializable_hash, status: :ok
      end
    end
  end
end

# UsersController
module Api
  module V1
    class UsersController < ApplicationController
    	skip_before_action :doorkeeper_authorize!
    	...
  	end
	end
end
```

Reply:
Fixed in https://github.com/gutakk/ruby-google-scraper/pull/19